packages/library-linter/src/linter.ts

```diff
@@ -0,0 +1,41 @@
+import { NamespaceType, Program, SymbolFlags, Type } from "@cadl-lang/compiler";
+import { reportDiagnostic } from "./lib.js";
+
+export function $onValidate(program: Program) {
+  const root = program.checker!.getGlobalNamespaceType();
+
+  validateNoExportAtRoot(program, root);
+}
+
+function validateNoExportAtRoot(program: Program, root: NamespaceType) {
+  function validateFor(items: Map<string, Type & { name?: string }>) {
```

Comment:
might consider calling this validateNoExports, so that the meaning of the invocations below is clearer